acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/tls/support/CertificateHelpers.java

```diff
@@ -0,0 +1,135 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.tests.tls.support;
+
+import tech.pegasys.ethsigner.tests.dsl.tls.TlsCertificateDefinition;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.StringJoiner;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+
+public class CertificateHelpers {
+
+  public static KeyManager[] createKeyManagers(final TlsCertificateDefinition certToPresent)
+      throws KeyStoreException, NoSuchAlgorithmException, CertificateException,
+          UnrecoverableKeyException {
+    if (certToPresent == null) {
+      return null;
+    }
+
+    final String password = certToPresent.getPassword();
+
+    final KeyStore clientCertStore = loadP12KeyStore(certToPresent.getPkcs12File(), password);
+
+    final KeyManagerFactory kmf = KeyManagerFactory.getInstance("PKIX");
+    kmf.init(clientCertStore, password.toCharArray());
+    return kmf.getKeyManagers();
+  }
+
+  public static TrustManagerFactory createTrustManagerFactory(
+      final TlsCertificateDefinition serverCert)
+      throws KeyStoreException, NoSuchAlgorithmException, CertificateException {
+
+    final KeyStore trustStore =
+        loadP12KeyStore(serverCert.getPkcs12File(), serverCert.getPassword());
+    final TrustManagerFactory trustManagerFactory =
+        TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+    trustManagerFactory.init(trustStore);
+
+    return trustManagerFactory;
+  }
+
+  public static KeyStore loadP12KeyStore(final File pkcsFile, final String password)
+      throws KeyStoreException, NoSuchAlgorithmException, CertificateException {
+    final KeyStore store = KeyStore.getInstance("pkcs12");
+    try (final InputStream keystoreStream = new FileInputStream(pkcsFile)) {
+      store.load(keystoreStream, password.toCharArray());
+    } catch (IOException e) {
+      throw new RuntimeException("Unable to load keystore.", e);
+    }
+    return store;
+  }
+
+  public static void populateFingerprintFile(
```

Comment:
can remove the UnrecoverableKeyException it's not needed

Reply:
done